mewarpx/mewarpx/utils_store/oracle_control.py

```diff
@@ -1,85 +1,112 @@
-        """Arguments:
+    def __init__(self, diag_steps, dt, first_data, override_bins=None,
+                 work_dir=None, run_script=None, save_dir=None, prefix=None,
+                 species_names=None, coarsening_factor=None):
+        """
+        Arguments:
             diag_steps (int): number of time steps in a diagnostic interval
-            dt (float): time of timestep
-            start_time (float): time in seconds that the simulation will start at
+            dt (float): time of timestep in seconds
+            first_data (float): time in seconds of the first data point used in
+                the regression
             override_bins (list): list of 3 lists that contains override variables
                 for determining prediction step parameters:
                 - list 0: list of start time bins
                 - list 1: list of possible interval values
                 - list 2: list of prediction time values
             work_dir (str): directory path where the runs are located, defaults
                 to current directory
-            save_dir (str): directory path to save generated prediction files
+            save_dir (str): directory path to save generated prediction files,
+                defaults to "work_dir/seed_density"
             prefix (str): prefix of run directories, defaults to "start"
             species_names (list): list of species names or identifying shared
-                substrings of species names, eg. electrons or Xe_ion
+                substrings of species names, eg. electrons or Xe_ion. The glob syntax
+                using species name would be
+                `{species_name}*_particle_density_{self.time_steps[i]:010d}.npy`
             coarsening_factor (int): the down-sampling integer factor along
                 each axis for the grid
         """
         if work_dir is None:
-            self.work_dir = os.curdir
+            self.work_dir = os.path.abspath(os.curdir)
         else:
             self.work_dir = work_dir
 
         self.output_file = os.path.join(self.work_dir, "oracle_control.txt")
 
         # remove output file if it exists so that the autorun script will exit
-        # if any unexpected erros are thrown in this script
+        # if any unexpected errors are thrown in this script
         if os.path.isfile(self.output_file):
             os.remove(self.output_file)
 
         # set time values
         self.diag_interval = diag_steps
         self.dt = dt
-        self.start_step = int(np.round(start_time / (dt * diag_steps))) * diag_steps
+        self.start_step = self._time_to_step(first_data)
 
         # set class variables
+        if run_script is None:
+            run_script = "run_simulation.py"
+        if prefix is None:
+            prefix = "start"
         if species_names is None:
             species_names = ["electron", "ion"]
+        if coarsening_factor is None:
+            coarsening_factor = 8
```

Comment:
I assume defaults are in here rather than function definition because the command line calls explicitly include None in some cases. That's fine, but a comment here would be helpful.
Also, add the defaults to the docstring for each argument; otherwise documentation won't show defaults at all.